app/user.go

```diff
@@ -1634,3 +1634,118 @@ func (a *App) RecordUserTermsOfServiceAction(userId, termsOfServiceId string, ac
+	var err *model.AppError
+
+	// Get learning group prefixes (ex. 'plg' or 'capstone')
+	// Learning group prefixes are used to identify private channels that
+	// were automatically generated for a learning group
+	//     Ex. If a private channel's name is 'plg-30', and the following function
+	//         returns 'plg' as a learning group prefix, then we can reasonably assert
+	//         that channel 'plg-30', was an automatically generated private channel
+	//         for a learning group of type 'plg'
+	if learningGroupPrefixes, err = a.GetLearningGroupPrefixes(teamId); err != nil {
+		// Return and exit from GetUserAnalytics with error
+		return nil, err
+	}
+
+	// Add learningGroupPrefixes to return data (always return learning group prefixes)
+	// Assert as type 'array of pointers to LearningGroupPrefix structs'
+	if returnData.LearningGroupPrefixes, assertionSuccess = interface{}(learningGroupPrefixes).([]*model.LearningGroupPrefix); !assertionSuccess {
+		// Return and exit from GetUserAnalytics with error
+		return nil, model.NewAppError("GetUserAnalytics", "api.user.get_user_analytics.learning_group_prefix_assertion_error", nil, "", http.StatusBadRequest)
+	}
+
+	// Manually check what data set is being requested
+	if name == "UserInteractions" {
+		if userInteractions := <-a.Srv.Store.User().GetUserInteractions(userId, teamId, learningGroupPrefixes); userInteractions.Err != nil {
+			// Return and exit from GetUserAnalytics with error
+			return nil, userInteractions.Err
+		} else {
+			// Add UserInteractions data to the returnData object
+			// Assert as type 'array of pointers to UserInteraction structs'
+			if returnData.UserInteractions, assertionSuccess = userInteractions.Data.([]*model.UserInteraction); !assertionSuccess {
+				// Return and exit from GetUserAnalytics with error
+				return nil, model.NewAppError("GetUserAnalytics", "api.user.get_user_analytics.user_interaction_assertion_error", nil, "", http.StatusBadRequest)
+			}
+		}
+	}
+
+	// Always return the user's learning groups
+	if userLearningGroups := <-a.Srv.Store.User().GetUserLearningGroups(userId, teamId, learningGroupPrefixes); userLearningGroups.Err != nil {
+		// Return and exit from GetUserAnalytics with error
+		return nil, userLearningGroups.Err
+	} else {
+		// Add LearningGroups data to the returnData object
+		// Assert as type 'array of pointers to LearningGroup structs'
+		if returnData.UserLearningGroups, assertionSuccess = userLearningGroups.Data.([]*model.LearningGroup); !assertionSuccess {
+			// Return and exit from GetUserAnalytics with error
+			return nil, model.NewAppError("GetUserAnalytics", "api.user.get_user_analytics.learning_group_assertion_error", nil, "", http.StatusBadRequest)
+		}
+	}
+
+	// Return returnData, which is of type UserAnalytics, and nil as error
+	return returnData, nil
+}
+
+// This function is used to fetch the learning group prefixes in the configuration file
+// for the selected mattermost team
```

Comment:
Maybe add just a little more here. Why might you want to do this? I _think_ the reason is that you'll use these prefixes to construct an SQL query to build the network graph. The reason I think this is that I grep'd through the source code a couple of times to find where this was called, and then to find the functions that its results were passed to. If you document this fact here, a future reader will know what they might break (and what to re-test) if they change this.

Reply:
NVM, I see that this is well documented below.

Reply:
Improved documentation.

Reply:
Oops, well improved it anyway, let me know if you think it's redundant.

Reply:
I think the new stuff helps too. Now it's all very clear.